tests/cleaner_tests/basic_function_tests/report_with_mask.py

```diff
@@ -119,7 +119,7 @@ def test_binary_removed(self):
         self.assertFileNotCollected('var/log/binary_test.tar.xz')
 
     def test_binaries_removed_reported(self):
-        self.assertOutputContains('\[removed .* unprocessable files\]')
+        self.assertOutputContains(r'\[removed .* unprocessable files\]')
```

Comment:
I have which might be a better idea, so tagging @TurboTurtle and @pmoravec to gather their thoughts.

On the `assertOutputContains`, `assertOutputNotContains`, `assertSosLogContains` maybe we could convert the input to a raw string and pass the raw string to the regexp method?
This way we could always use a regular string as an input to those functions rather than remembering to pass a raw string.

Example on `assertOutputContains`:
```
content_raw = fr"{content}"
found = re.search(fr"(.*)?{content_raw}(.*)?", self.cmd_output.stdout + self.cmd_output.stderr)
```
Hope it makes sense.

Reply:
Eah that makes sense to me.

Reply:
Yep, makes sense here.

Reply:
this doesn't work, we still need the change to add the `r` before the string in the function, otherwise we get loads of these messages

```
tests/cleaner_tests/basic_function_tests/report_with_mask.py:122: SyntaxWarning: invalid escape sequence '\['
  self.assertOutputContains('\[removed .* unprocessable files\]'
```

below is the diff I had

```diff
diff --git a/tests/cleaner_tests/basic_function_tests/report_with_mask.py b/tests/cleaner_tests/basic_function_tests/report_with_mask.py
index bb72cdde..3c860ba5 100644
--- a/tests/cleaner_tests/basic_function_tests/report_with_mask.py
+++ b/tests/cleaner_tests/basic_function_tests/report_with_mask.py
@@ -119,7 +119,7 @@ class DefaultRemoveBinaryFilesTest(StageTwoReportTest):
         self.assertFileNotCollected('var/log/binary_test.tar.xz')
 
     def test_binaries_removed_reported(self):
-        self.assertOutputContains(r'\[removed .* unprocessable files\]')
+        self.assertOutputContains('\[removed .* unprocessable files\]')
 
 
 class KeepBinaryFilesTest(StageTwoReportTest):
diff --git a/tests/sos_tests.py b/tests/sos_tests.py
index 1fb31875..fd833183 100644
--- a/tests/sos_tests.py
+++ b/tests/sos_tests.py
@@ -333,7 +333,8 @@ class BaseSoSTest(Test):
         :param content:  The string that should not be in stdout
         :type content:  ``str``
         """
-        found = re.search(r"(.*)?%s(.*)?" % content, self.cmd_output.stdout + self.cmd_output.stderr)
+        content_raw = fr"{content}"
+        found = re.search(fr"(.*)?{content}(.*)?", self.cmd_output.stdout + self.cmd_output.stderr)
         assert found, "Content string '%s' not in output" % content
 
     def assertOutputNotContains(self, content):
@@ -342,7 +343,8 @@ class BaseSoSTest(Test):
         :param content:  The string that should not be in stdout
         :type content:  ``str``
         """
-        found = re.search(r"(.*)?%s(.*)?" % content, self.cmd_output.stdout + self.cmd_output.stderr)
+        content_raw = fr"{content}"
+        found = re.search(fr"(.*)?{content}(.*)?", self.cmd_output.stdout + self.cmd_output.stderr)
         assert not found, "String '%s' present in stdout" % content
 
```

Reply:
You need to remove the escapes that the error complaints about like `\[` and `\(` as they are not valid escape sequences anymore, I think there are also more warnings for different characters and they need to be removed as well.

Reply:
After some testing and talking with @arif-ali , seems that I was wrong on this :) one -needs- to escape the \[ and \( (and others..) in order for the regexp to work properly.

The alternative for us is to sanitize the input on all the functions, similar to my comment, instead of converting to raw we would have to verify if there are any non-escaped characters, escape them, and input that into the `re.` functions. But this is already some extra work and I think we should proceed and just convert those strings to raw where necessary, as well as doing this for all of the future string matching that we need.